src/TrigScint/TrigScintQIEDigiProducer.cxx

```diff
@@ -0,0 +1,148 @@
+    stripsPerArray_   =
+      parameters.getParameter< int >("number_of_strips");
+    numberOfArrays_   =
+      parameters.getParameter< int >("number_of_arrays");
+    meanNoise_        =
+      parameters.getParameter< double >("mean_noise");
+    mevPerMip_        =
+      parameters.getParameter< double >("mev_per_mip");
+    pePerMip_         =
+      parameters.getParameter< double >("pe_per_mip");
+    inputCollection_  =
+      parameters.getParameter< std::string >("input_collection");
+    inputPassName_    =
+      parameters.getParameter< std::string >("input_pass_name" );
+    outputCollection_ =
+      parameters.getParameter< std::string >("output_collection");
+    verbose_          =
+      parameters.getParameter< bool >("verbose");
+
+    // QIE specific parameters initialization
+    maxts_ 	      	=
+      parameters.getParameter< int >("maxts");
+    toff_overall_     	=
+      parameters.getParameter< double >("toff_overall");
+    input_pulse_shape_ 	=
+      parameters.getParameter< std::string >("input_pulse_shape");
+    tdc_thr 		=
+      parameters.getParameter< double >("tdc_thr");
+    pedestal 		=
+      parameters.getParameter< double >("pedestal");
+    elec_noise 		=
+      parameters.getParameter< double >("elec_noise");
+    sipm_gain 		=
+      parameters.getParameter< double >("sipm_gain");
+    s_freq 		=
+      parameters.getParameter< double >("qie_sf");
+      
+    if ( input_pulse_shape_ == "Expo") {
+      pulse_params.clear();
+      pulse_params.push_back
+	(parameters.getParameter< double >("expo_k"));
+      pulse_params.push_back
+	(parameters.getParameter< double >("expo_tmax"));
+    }
+  }
+
+  void TrigScintQIEDigiProducer::produce(Event& event) {
+
+    // Need to handle seeding on the first event
+    if (!hasSeed()) {
+      const auto& rseed = getCondition<RandomNumberSeedService>
+	(RandomNumberSeedService::CONDITIONS_OBJECT_NAME);
+      const auto& rseed2 =getCondition<RandomNumberSeedService>
+	(RandomNumberSeedService::CONDITIONS_OBJECT_NAME);
+        
+      random_ =
+	std::make_unique<TRandom3>(rseed.getSeed(outputCollection_));
+      smq = new SimQIE
+	(pedestal,elec_noise,rseed2.getSeed(outputCollection_+"SimQIE"));
+      smq->SetGain(sipm_gain);
+      smq->SetFreq(s_freq);
+      smq->SetNTimeSamples(maxts_);
+      smq->SetTDCThreshold(tdc_thr);
+    }
+
+    // To simulate multiple pulses coming at different times, SiPMS
+    float TrueEdep[stripsPerArray_];
+    Expo* ex[stripsPerArray_]={0};
+    for(int i=0;i<stripsPerArray_;i++){
+      ex[i] = new Expo(pulse_params[0],pulse_params[1]);
+      TrueEdep[i]=0;
+    }
+
+    // loop over sim hits and aggregate energy depositions for each detID
+    const auto simHits{event.getCollection< SimCalorimeterHit >
+	(inputCollection_,inputPassName_)};
+
+    for (const auto& simHit : simHits) {
+
+      TrigScintID id(simHit.getID());
+
+      if (verbose_) {
+	std::cout << id << std::endl;
```

Comment:
Here, move to using `ldmx_log(debug) << "Processing sim hit " << id ; ` (or something) instead of `std::cout`